### PR TITLE
Adding ArgoCD terminology and references

### PR DESCRIPTION
**Pull request by murillodigital**

### CODE REVIEW

File: docs/vocabulary.md [hunk 1]
```diff
@@ -337,7 +361,8 @@ pipeline in corresponding documentation.
 
 | Project            |          |          |          |          |         |              |
 |--------------------|----------|----------|----------|----------|---------|--------------|
-| **CircleCI**       | N/A      | Step     | Job      | Workflow | Trigger | Executor     |
+| **ArgoCD**         | Sync Wave         | Sync Phase  | Sync      |N/A      | Event         |  Application Controller            |
+| **CircleCI**       | N/A      | Step     | Job      | Workflow | Trigger | Executor     |                    |          |          |          |          |         |              |
```
Comment (zxiiro) [inline docs/vocabulary.md:365]: Looks like there's some extra |s here that was accidently added?

File: docs/vocabulary.md [hunk 2]
```diff
@@ -51,6 +52,29 @@ humans to communicate and collaborate better.
...
 
+### ArgoCD
+
+[ArgoCD](https://argoproj.github.io/argo-cd/) is a declarative, GitOps continuous delivery tool for Kubernetes.
+
+Some of the core ArgoCD concept are listed below: [[16]]
+
+- **Application**: A group of Kubernetes resources as defined by a manifest. This is a Custom Resource Definition (CRD).
+- **Application controller** [[19]]: Kubernetes controller which continuously monitors running applications and compares live vs. desired state.
+- **Application source type**: Which Tool is used to build the application.
```
Comment (zxiiro) [inline docs/vocabulary.md:63]: Tool should be lowercase t.

**Review by zxiiro — CHANGES_REQUESTED** [review]

File: docs/vocabulary.md (hunk 1 above)
Comment (murillodigital) [reply docs/vocabulary.md]: Thanks for catching that - indeed accidental (IDE introduced?)

File: docs/vocabulary.md (hunk 2 above)
Comment (murillodigital) [reply docs/vocabulary.md]: Switched to lower case, I believe the intent of the uppercase (as used in the original reference) is to denote that the term is also one defined in the list, I don't think lower case makes its any less comprehensible though, and it does _look_ better.

**Review by zxiiro — APPROVED** [review]
Looks good to me, thanks!

**Comment by murillodigital** [comment]
thanks @zxiiro ! Anything else pending on me towards getting this merged?

**Comment by zxiiro** [comment]
> thanks @zxiiro ! Anything else pending on me towards getting this merged?

Nothing I can think of. Let's see if any others want to review before merging. It's holiday season so likely folks are taking time off. Let's give them another week and if not I'm happy to merge the patch.

**Review by fdegir — APPROVED** [review]
Thanks a lot for adding ArgoCD terminology and references!

**Pull request merged**